src/btree/col_srch.c

```diff
@@ -328,8 +339,12 @@ __wt_col_search(
             cbt->compare = 0;
         else if (recno < cbt->recno)
             cbt->compare = 1;
-        else
+        else {
             cbt->compare = -1;
+            /* Special case for FLCS: indicate there is a row larger than the searched-for key. */
+            if (greater_recno_exists_flcs)
+                cbt->compare = 1;
```

Comment:
Here, we are in the case where the insert list is not `NULL`. I have not been able to prove if the same happens when the insert list is `NULL`.